statistics/statistics.rb

```diff
@@ -0,0 +1,36 @@
+module Statistics
+  def top_readers(orders, quantity = 1)
+    get_statistics(orders, 'reader', quantity)
+  end
+
+  def top_books(orders, quantity = 1)
+    get_statistics(orders, 'book', quantity)
+  end
+
+  def unique_readers_number(orders, quantity = 3)
+    top_books = get_statistics(orders, 'book', quantity).map!(&:title)
+    orders.select { |order| order.reader.name if top_books.include? order.book.title }.compact.size
+  end
+
+  private
+
+  def get_statistics(orders, group_param, quantity)
+    validate_orders(orders, quantity)
+
+    sort_orders(orders, group_param, quantity).to_h.keys
+  end
+
+  def validate_orders(entity, quantity)
+    check_empty(entity)
+    check_quantity(entity, quantity)
+  end
+
+  def sort_orders(orders, group_param, quantity)
```

Comment:
1) maybe you should send to group_param `:reader`/`:book`
2) what if i send to sort_orders not `reader`/`book` string